server/simulation/vehicles.js

```diff
@@ -15,7 +15,7 @@ const randomBid = (origin, pickup, dropoff) => {
   const dropoffPoint = turf.point([parseFloat(dropoff.long), parseFloat(dropoff.lat)]);
   const distanceOriginToPickup = turf.distance(originPoint, pickupPoint);
   const distancePickupToDelivery = turf.distance(pickupPoint, dropoffPoint);
-  const price = (distanceOriginToPickup+distancePickupToDelivery).toFixed(2);
+  const price = parseFloat((distanceOriginToPickup+distancePickupToDelivery).toFixed(2));
```

Comment:
I see the value of what you did here... but this task is about testing the current state of the app.

The change you introduced here changes the type of data the app returns and may have unintended effects here and in Missions. In fact, it will change the way prices are displayed in Missions' UI, which we do not want.

In other words, this is a *breaking change* and not something that can be introduced as part of another unrelated pull request. It needs to be considered and tested carefully and missions would need to be updated accordingly... just a part of the reality of developing open source software that is used by many different people. You need to make sure you do not accidentally break other people's software who may rely on that being a string (even if it shouldn't be).

But yes, this does seem like a good change and you uncovered an actual issue here.

What I suggest is that you create a separate pull request to Missions that expects to receive a float in the price field and calls `toFixed(2)` when it wants to display that value in the UI. Then make the change you suggested to this method in a separate pull request to Mission Control. Then finally we'll do this pull request here that only handles testing.

I know it sounds like a hassle, but I think it should take you less time to do it then it took me to write the above ;)

Reply:
I had to use an alternative solution for trimming the decimals, but here is the pull request: https://github.com/DAVFoundation/missions/pull/40 😀